beacon_node/beacon_chain/src/beacon_block_streamer.rs

```diff
@@ -0,0 +1,719 @@
+
+            match load_result {
+                Ok(Some(LoadedBeaconBlock::Blinded(blinded_block))) => {
+                    match blinded_block
+                        .message()
+                        .execution_payload()
+                        .map(|payload| payload.to_execution_payload_header())
+                    {
+                        Ok(header) => {
+                            let block_parts = BlockParts::new(blinded_block, header);
+                            if block_parts.slot() <= self.finalized_slot {
+                                // this is a by_range request
+                                by_range_blocks.push(block_parts);
+                            } else {
+                                // this is a by_hash request
+                                by_hash
+                                    .push_block_parts(block_parts, &self.beacon_chain.log)
+                                    .await;
+                                requests.insert(root, by_hash.clone());
+                            }
+                        }
+                        Err(_) => {
+                            no_request
+                                .push_block_result(
+                                    root,
+                                    Err(BeaconChainError::BlockVariantLacksExecutionPayload(root)),
+                                    &self.beacon_chain.log,
+                                )
+                                .await;
+                            requests.insert(root, no_request.clone());
+                        }
+                    }
+                }
+                // no request when there's an error, or the block doesn't exist, or we already have the full block
+                no_request_load_result => {
+                    let block_result = match no_request_load_result {
+                        Err(e) => Err(e),
+                        Ok(None) => Ok(None),
+                        Ok(Some(LoadedBeaconBlock::Full(full_block))) => Ok(Some(full_block)),
+                        // unreachable due to the match statement above
+                        Ok(Some(LoadedBeaconBlock::Blinded(_))) => unreachable!(),
+                    };
+                    no_request
+                        .push_block_result(root, block_result, &self.beacon_chain.log)
+                        .await;
+                    requests.insert(root, no_request.clone());
+                }
+            }
+        }
+
+        // Now deal with the by_range requests. Sort them in order of increasing slot
+        let mut by_range = EngineRequest::<T::EthSpec>::new_by_range();
+        by_range_blocks.sort_by(|block_parts_a, block_parts_b| {
+            // this unwrap shouldn't occur as slot is never u64::NAN
+            block_parts_a
+                .slot()
+                .partial_cmp(&block_parts_b.slot())
+                .unwrap_or(std::cmp::Ordering::Equal)
+        });
```

Comment:
`Slot` implements `Ord` so we can use `sort_by_key`:

```suggestion
        by_range_blocks.sort_by_key(|block_parts| block_parts.slot());
```